### PR TITLE
Update importlib/test_importlib from CPython 3.10.6

### PR DESCRIPTION
**Pull request by youknowone**
~I have no idea whats going on its failure.~

blocked by ~#3863~, #3961

### CODE REVIEW

**Comment by fanninpm** [comment]
Directory tree in RustPython (as it stands right now):

```text
Lib/importlib/
├── __init__.py
├── _bootstrap.py
├── _bootstrap_external.py
├── abc.py
├── machinery.py
├── metadata.py
├── resources.py
└── util.py
```

Directory tree in CPython 3.10:

```text
Lib/importlib/
├── metadata/
│   ├── __init__.py
│   ├── _adapters.py
│   ├── _collections.py
│   ├── _functools.py
│   ├── _itertools.py
│   ├── _meta.py
│   └── _text.py
├── __init__.py
├── _abc.py
├── _adapters.py
├── _bootstrap.py
├── _bootstrap_external.py
├── _common.py
├── abc.py
├── machinery.py
├── readers.py
├── resources.py
└── util.py
```

**Comment by youknowone** [comment]
I am sorry, it was my mistake to miss new files in commit. The error is very unexpected:

```
Traceback (most recent call last):
  File "/Users/user/Projects/RustPython/vm/pylib-crate/Lib/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/Users/user/Projects/RustPython/vm/pylib-crate/Lib/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "_frozen_importlib", line 1129, in __import__
  File "_frozen_importlib", line 1050, in _gcd_import
  File "_frozen_importlib", line 1027, in _find_and_load
  File "_frozen_importlib", line 1027, in _find_and_load
  File "_frozen_importlib", line 1006, in _find_and_load_unlocked
  File "_frozen_importlib", line 703, in _load_unlocked
  File "_frozen_importlib", line 701, in _load_unlocked
  File "_frozen_importlib", line 694, in _load_unlocked
  File "_frozen_importlib", line 688, in _load_unlocked
  File "_frozen_importlib_external", line 879, in exec_module
  File "_frozen_importlib_external", line 1017, in get_code
  File "_frozen_importlib_external", line 948, in source_to_code
  File "_frozen_importlib", line 241, in _call_with_frames_removed
  File "/Users/user/Projects/RustPython/vm/pylib-crate/Lib/test/test_importlib/test_util.py", line 1
    from . import util
     ^
SyntaxError: Got unexpected token  at line 1 column 2
from . import util
 ^

```

**Comment by youknowone** [comment]
Thank you, that explains the diff. It means we don't support bom for files.

**Comment by fanninpm** [comment]
[CPython has known about this issue for quite some time.](https://github.com/python/cpython/issues/46635)

File: Lib/test/test_importlib/test_util.py [hunk 1]
```diff
@@ -1,10 +1,9 @@
-from . import util
+﻿from . import util
```
[the remark this answers is not in the document]
Comment (youknowone) [reply Lib/test/test_importlib/test_util.py:1]: Unfortunately, the file you linked includes BOM. So this is came from CPython

[the remark this answers is not in the document]
Comment (youknowone) [reply Lib/test/test_importlib/test_util.py:1]: Ok, they at least don't have one for master. So this is not intended.

File: Lib/test/test_importlib/util.py [hunk 2]
```diff
@@ -295,7 +307,7 @@ def writes_bytecode_files(fxn):
     """Decorator to protect sys.dont_write_bytecode from mutation and to skip
     tests that require it to be set to False."""
     if sys.dont_write_bytecode:
-        return lambda *args, **kwargs: None
+        return unittest.skip(fxn)
```
Comment (youknowone) [inline Lib/test/test_importlib/util.py:310]: this change is submitted to https://github.com/python/cpython/pull/95972

**Pull request merged**